devicetypes/technisat/technisat-series-switch-child.src/technisat-series-switch-child.groovy

```diff
@@ -0,0 +1,106 @@
+/**
+ *	Copyright 2021 TechniSat
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *	in compliance with the License. You may obtain a copy of the License at:
+ *
+ *		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+metadata {
+	definition (name: "TechniSat Series switch child", namespace: "TechniSat", author: "TechniSat", vid: "generic-switch-power-energy",
+				mnmn: "SmartThings") {
+		capability "Energy Meter"
+		capability "Switch"
+		capability "Power Meter"
+		capability "Refresh"
+		capability "Health Check"
+	}
+}
+
+def installed() {
+	log.debug "installed()"
+	initialize()
+}
+
+def updated() {
+	log.debug "updated()"  
+	initialize()
+}
+
+def initialize() {
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	refresh()
+}
+
+def handleMeterReport(cmd) {
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			sendEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+		} else if (cmd.scale == 1) {
+			sendEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
+		} else if (cmd.scale == 2) {
+			sendEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
+}
```

Comment:
This is essentially duplicated from the parent DTH. I believe you'd save a lot of code duplication if you just rewrote your main DTH to use the "smartthings/Child Metering Switch" DTH.

You've already written your zwave handlers in the main DTH to handle an endpoint argument, shouldn't be too much work to just have them create the child's event when the endpoint is "2"